cloudbuild-test.yaml

```diff
@@ -0,0 +1,20 @@
+steps:
+  - name: 'gcr.io/cloud-builders/gcloud'
+    script: |
+      #!/usr/bin/env bash
+      set -e
+      REPO_TOKEN="$(gcloud auth print-access-token)"
+      cat > ~/.npmrc <<EOL
+      registry=https://us-npm.pkg.dev/artifact-foundry-prod/ah-3p-staging-npm/
+      //us-npm.pkg.dev/artifact-foundry-prod/ah-3p-staging-npm/:always-auth=true
+      //us-npm.pkg.dev/artifact-foundry-prod/ah-3p-staging-npm/:_authToken=${REPO_TOKEN}
+      EOL
+  - name: gcr.io/cloud-builders/bazel:5.4.0
+    script: |
+      #!/usr/bin/env bash
+      set -e
+      bazel run @nodejs//:yarn config set registry https://us-npm.pkg.dev/artifact-foundry-prod/ah-3p-staging-npm/
+      ./scripts/run_tests
```

Comment:
Does the `./scripts/run` script still work when run locally? Or does it have to use this registry, now that the yarn lock has been updated?

Reply:
It have to use the new registry and local tests would not be possible. This is the discussion we had when we decided to keep single yarn.lock file. Ideal, user can use npm/yarn to update yarn.lock to point to yarnpkg.com registry and then run tests.

Reply:
Wait, does this mean that local development isn't possible either? Is it no longer possible to run `yarn install` after cloning the repo? We need a workaround if this is the case!

Reply:
Yes. Local development is no longer possible. To run local development, users will have to delete yarn.lock file and run install again. 

One workaround i can think about is to keep both versions of yarn.lock files. something like yarn.lock and yarn-local.lock. and users can change name before running yarn install. However, I do not support this solution as this will become problem very quickly trying to keep both files in sync. 

For authorization error you need o update the .npmrc file to same as the once created by `cloudbuild-test.yaml`

Reply:
So we definitely need to be able to keep developing... not quite sure why you think it would be OK to merge this.

Also ideally it needs to be done in a way that uses the same dependencies as when airlock is called

Some potential solutions:
* Have the tests run a replace of `registry.yarnpkg.com/` with `us-npm.pkg.dev/artifact-foundry-prod/ah-3p-staging-npm/` in the lockfile at test / publish time.
* Find a way for us to be able to access the airlock registry from our cloudtop machines.
* (We shouldn't do this due to non-determinism of installs) not write a yarn lockfile, and generate it at install

Reply:
This has now been resolved by the addition of a `./scripts/run_tests_local`

@Tuseeq1  could you rename `./scripts/run_tests_local` to `./scripts/run_tests`, and `./scripts/run_tests` to `./scripts/run_tests_on_cloudbuild`? This way we don't need to update the internal development guide.